packages/net-stubbing/lib/server/util.ts

```diff
@@ -208,3 +209,22 @@ export function mergeDeletedHeaders (before: CyHttpMessages.BaseMessage, after:
     !after.headers[k] && delete before.headers[k]
   }
 }
+
+type BodyEncoding = 'utf8' | 'binary' | null
+
+export function getBodyEncoding (req: CyHttpMessages.IncomingRequest): BodyEncoding {
+  if (!req.body) {
+    return null
+  }
+
+  // a simple heuristic for detecting UTF8 encoded requests
+  if (req.headers && req.headers['content-type']) {
+    if (req.headers['content-type'].includes('charset=UTF-8')) {
```

Comment:
I think this check is not comprehensive enough, check the RFC, there are several other valid ways to write this: https://tools.ietf.org/html/rfc7231#section-3.1.1.1

Reply:
the above is how Chrome requests arrive by default, I can write a regexp, did not want this to be expensive

Reply:
Let's err on the side of correctness - like what if a user HTTP request library sets `charset`?

Reply:
would these be enough?
```
const types = [
        'text/html;charset=utf-8',
        'text/html;charset=UTF-8',
        'Text/HTML;Charset="utf-8"',
        'text/html; charset="utf-8"'
      ]
```

we are only looking at charset

Reply:
Why use a list at all? Do `/charset="?utf\-8"?/i.test(str)`. I promise that the microseconds wasted, if any, by using regex, will be worth it when we don't have to deal with weird bug reports in the future due to this.